save-backend/src/main/kotlin/com/saveourtool/save/backend/service/UserDetailsService.kt

```diff
@@ -183,34 +185,37 @@ class UserDetailsService(
 
     /**
      * @param name name of user
+     * @param status new status of user
      * @param authentication
      * @return UserSaveStatus
      */
     @Transactional
     fun deleteUser(
         name: String,
+        status: UserStatus,
         authentication: Authentication,
     ): UserSaveStatus {
         val user: User = userRepository.findByName(name).orNotFound()
-        val newName = "Deleted-${user.id}"
-        if (user.id == authentication.userId()) {
-            userRepository.deleteHighLevelName(user.name)
-            userRepository.saveHighLevelName(newName)
-            userRepository.save(user.apply {
-                this.name = newName
-                this.status = UserStatus.DELETED
-                this.avatar = null
-                this.company = null
-                this.twitter = null
-                this.email = null
-                this.gitHub = null
-                this.linkedin = null
-                this.location = null
-            })
-        } else {
+        val newName = if (status == UserStatus.DELETED) "Deleted-${user.id}" else "Banned-${user.id}"
```

Comment:
I'm not sure, it's a good to change username when user is banned -- we won't able to unban it if the username will be occupied